Remove redundant Python 2.6 stuff

hugovk opened the pull request:
Same as https://github.com/requests/requests/pull/4326 but for the proposed/3.0.0 branch.

The good news is nearly everything was already in this branch from https://github.com/requests/requests/pull/4118, except for tidying up some OrderedDict compat stuff.

Also, pytest>=3.0.0 is required for the Python 3.3 build, otherwise:
```
py.test -n 8 --boxed --junitxml=report.xml
Traceback (most recent call last):
  File "/home/travis/virtualenv/python3.3.6/bin/py.test", line 11, in <module>
    sys.exit(main())
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/config.py", line 39, in main
    config = _prepareconfig(args, plugins)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/config.py", line 118, in _prepareconfig
    pluginmanager=pluginmanager, args=args)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 724, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 338, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 333, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 595, in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 249, in _wrapped_call
    wrap_controller.send(call_outcome)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/helpconfig.py", line 28, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 278, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 264, in __init__
    self.result = func()
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 596, in execute
    res = hook_impl.function(*args)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/config.py", line 861, in pytest_cmdline_parse
    self.parse(args)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/config.py", line 966, in parse
    self._preparse(args, addopts=addopts)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/config.py", line 927, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/_pytest/vendored_packages/pluggy.py", line 501, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pkg_resources/__init__.py", line 2404, in load
    self.require(*args, **kwargs)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pkg_resources/__init__.py", line 2427, in require
    items = working_set.resolve(reqs, env, installer, extras=self.extras)
  File "/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages/pkg_resources/__init__.py", line 872, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.VersionConflict: (pytest 2.9.2 (/home/travis/virtualenv/python3.3.6/lib/python3.3/site-packages), Requirement.parse('pytest>=3.0.0'))
```
https://travis-ci.org/hugovk/requests/jobs/284048190

The other CI failures are related to cookies and also occur in latest master and fresh builds of proposed/3.0.0.

nateprewitt commented on requirements.txt:
```diff
@@ -1,5 +1,5 @@
 -e .[socks]
-pytest>=2.8.0
+pytest>=3.0.0
```
We've removed the requirement for python 3.3 in the 3.0.0 branch, so lets save this change for a separate PR if it's really needed. (Although, I think the requirements.txt was removed on master and that should now be reflected on this branch)

hugovk replied:
I've removed this commit and rebased onto current proposed/3.0.0 and update AUTHORS.rst on this branch.

hugovk commented:
I've removed that commit and rebased onto current proposed/3.0.0 and update AUTHORS.rst on this branch.

> We could also use a CHANGELOG entry stating that the OrderedDict import no longer exists in compat.py.

@nateprewitt Is the changelog in https://github.com/requests/requests/blob/proposed/3.0.0/3.0-HISTORY.rst?

nateprewitt commented:
@hugovk yep, that'd be the place.

nateprewitt commented on requests/utils.py:
```diff
@@ -224,11 +224,11 @@ def from_key_val_list(value):
 
         >>> from_key_val_list([('key', 'val')])
-        OrderedDict([('key', 'val')])
+        collections.OrderedDict([('key', 'val')])
```
It looks like this is a cli example. We don't want the "collection" lead here. The interpreter is just showing the returned objects repr. Same with the line below this.

hugovk commented:
@nateprewitt CLI example updated and changelog added.

nateprewitt commented:
Alright, I think we're set here. Thanks for again for putting this together @hugovk!
